src/java.base/share/classes/java/util/UUID.java

```diff
@@ -181,6 +181,70 @@ public static UUID nameUUIDFromBytes(byte[] name) {
         return new UUID(md5Bytes);
     }
 
+    /**
+     * Static factory to create a version 7 (time-based) {@code UUID} with the current
+     * Unix timestamp in milliseconds.
+     *
+     * The {@code UUID} embeds the current Unix Epoch timestamp in milliseconds using
+     * {@link System#currentTimeMillis()} into the first 6 bytes, sets the version and
+     * variant bits as per the specification and fills the remaining bytes with random
+     * data from a cryptographically strong pseudo-random number generator.
+     *
+     * @return a {@code UUID} generated with the current Unix timestamp
+     *
+     * @spec RFC 9562
```

Comment:
The sections of the RFC describing the required monotonicity seem to require measures be taken in case of leap seconds and other clock setting cases.  System.currentTimeMillis() does not ensure a monotonic time.
Something using `nanoTime` could do better since nanoTime is monotonic.
```
  // Record a fixed point in nano-time corresponding to a fix point in System.currentTimeMillis()
  // and use the delta from the current nano-time as the current nano-time.
  private static final long ORIGIN_MS = System.currentTimeMillis();
  private static final long ORIGIN_NS = System.nanoTime();
  
  private static long monotonicMS() {
      return  ORIGIN_MS + (System.nanoTime() - ORIGIN_NS) / 1_000_000;
  }